packages/message-parser/src/grammar.pegjs

```diff
@@ -77,206 +82,154 @@ Code_line
+/**
+ *
+ * KaTex
+ * e.g: \[ KATEX_HERE \] OR $$ KATEX_HERE $$
+ * $$x = \begin{cases}
+ * a &\text{if } b \\
+ * c &\text{if } d
+ * \end{cases}$$
+ *
+ */
+Katex = katexStart content:$(!katexEnd .)* katexEnd { return katex(content); }
 
-Bold_Content
-  = text:(References / Italic / Strikethrough / Line / AnyBold)+ {
-      return bold(reducePlainTexts(text));
-    }
+katexStart
+  = & { return options.katex?.parenthesisSyntax; } "\\["
+  / & { return options.katex?.dollarSyntax; } "$$"
 
-/* ~~Mistaken text.~~ */
-/* ~Mistaken text.~ */
-Strikethrough
-  = [\x7E] [\x7E] s:Strikethrough_Content [\x7E] [\x7E] { return s; }
-  / [\x7E] s:Strikethrough_Content [\x7E] { return s; }
+katexEnd
+  = & { return options.katex?.parenthesisSyntax; } "\\]"
+  / & { return options.katex?.dollarSyntax; } "$$"
 
-Strikethrough_Content
-  = text:(References / Italic / Bold / Line / AnyStrike)+ {
-      return strike(reducePlainTexts(text));
+inlineKatex
+  = inlineKatexStart content:$(!inlineKatexEnd .)* inlineKatexEnd {
+      return inlineKatex(content);
     }
 
-AnyBold = t:[^\x0a\* ] { return plain(t); }
-
-AnyStrike = t:[^\x0a\~ ] { return plain(t); }
-
-AnyItalic = t:[^\x0a\_ ] { return plain(t); }
+inlineKatexStart
+  = & { return options.katex?.parenthesisSyntax; } "\\("
+  / & { return options.katex?.dollarSyntax; } "$"
 
-InlineCode = "`" text:$InlineCode__+ "`" { return inlineCode(plain(text)); }
+inlineKatexEnd
+  = & { return options.katex?.parenthesisSyntax; } "\\)"
+  / & { return options.katex?.dollarSyntax; } "$"
 
-InlineCode__ = $(!"`" !"\n" $:.)
+/**
+ *
+ * LineBreak
+ * e.g: \n
+ *
+*/
+LineBreak = Space* EndOfLine { return lineBreak(); }
 
-FilePath = $(URL_scheme URL_body+)
+/**
+ *
+ * Paragraph
+ * e.g: This is a paragraph
+*/
+Paragraph = value:Inline { return paragraph(value); }
 
-LinkTitle = text:(Emphasis / Line / Whitespace) { return text; }
+/**
+ *
+ * Inline
+ *
+*/
+Inline = value:(InlineItem / Any)+ EndOfLine? { return reducePlainTexts(value); }
+
+InlineItem = Whitespace
+  / References
+  / AutolinkedPhone
+  / AutolinkedEmail
+  / AutolinkedURL
+  / Emphasis
+  / UserMention
+  / ChannelMention
+  / Emoji
+  / InlineCode
+  / Image
+  / emoticon
+  / color
+  / inlineKatex
+  / Escaped
 
-LinkRef
-  = text:(URL / FilePath / p:Phone { return 'tel:' + p.number; }) {
-      return text;
-    }
+/**
+ *
+ * URL
+ * e.g:
+ * Reference: [Rocket.Chat Website](https://rocket.chat), [](https://rocket.chat), <rocket.chat|Rocket.Chat Website>
+ * Image: ![](https://rocket.chat/logo.png)
+ *
+ */
+References
+  = "[" title:LinkTitle* "](" href:LinkRef ")" { return title.length ? link(href, reducePlainTexts(title)) : link(href); }
+  / "<" href:LinkRef "|" title:LinkTitle2 ">" { return link(href, [plain(title)]); }
 
-Image
-  = "![](" href:LinkRef ")" { return image(href); }
-  / "![" title:LinkTitle "](" href:LinkRef ")" { return image(href, title); }
+LinkTitle = @(Emphasis / Line / Whitespace)
```

Comment:
It's the only item in sequence, not even `@` is needed

Reply:
Why not? The `LinkTitle` should get the  returned value, to make it work there are two options:

`LinkTitle = text:(Emphasis / Line / Whitespace) { return text; }`
OR as it is:
`LinkTitle = @(Emphasis / Line / Whitespace)`

Just the syntax is different, right? Is there any major differences/trade-offs/performance impact between them?